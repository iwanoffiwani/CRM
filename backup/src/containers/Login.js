import React, { Component } from 'react'
import { connect } from 'react-redux'
import { errorCreator, userCreator } from '../redux/actions/'
import store from '../redux/store'
import axios from 'axios'
import jwt_decode from 'jwt-decode'
import setAuthToken from '../shared/setAuthToken'
import history from '../history/'
import { makeStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import Typography from '@material-ui/core/Typography'
import FormGroup from '@material-ui/core/FormGroup'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'

const useStyles = makeStyles({
  container: {
    display: 'flex',
    alignItems: 'center',
    minHeight: '100vh',
    marginTop: '-5%'
  },
  submit: {
    marginTop: '2em'
  }
})

const MyContainer = ({ children }) => {
  const classes = useStyles()

  return (
    <Container maxWidth='xs' className={classes.container}>
      {children}
    </Container>
  )
}

const MySubmitButton = () => {
  const classes = useStyles()

  return (
    <Button
    type='submit'
    size='medium'
    color='primary'
    variant='contained'
    className={classes.submit}
    >Войти</Button>
  )
}

class Login extends Component {
  constructor() {
    super()

    this.changeHandler = this.changeHandler.bind(this)
    this.submitHandler = this.submitHandler.bind(this)

    this.state = {
      login: '',
      password: '',
      error: ''
    }
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.auth) 
      return this.props.history.push('/')

    if (nextProps.error)
      return this.setState({
        error: nextProps.error.message
      })
  }

  componentDidMount() {
    if(this.props.auth) {
        this.props.history.push('/');
    }
  }

  changeHandler(e) {
    return this.setState({
      [e.target.name]: e.target.value
    })
  }

  submitHandler(e) {
    e.preventDefault()

    const user = this.state

    axios.post('/api/auth/login', user)
      .then(res => {
        const { token } = res.data
        const decoded = jwt_decode(token)

        localStorage.setItem('jwtToken', token)
        setAuthToken(token)

        store.dispatch(userCreator(decoded))

        return history.push('/')
      })
      .catch(err => store.dispatch(errorCreator(err.response.data)))
  }
  
  render() {
    return (
      <MyContainer>
        <form style={{ width: '100%' }} onSubmit={this.submitHandler}>
          {
            this.state.error ? 
            <Typography
            color='error'
            variant='caption'
            >{this.state.error}</Typography> : false
          }
          <Typography
          color='textPrimary'
          variant='h5'
          >Вход в систему</Typography>
          <FormGroup>
            <TextField
            type='text'
            name='login'
            label='Логин'
            error={this.state.error}
            value={this.state.login}
            margin='normal'
            variant='outlined'
            onChange={this.changeHandler}
            placeholder='Введите ваш логин'
            />
          </FormGroup>
          <FormGroup>
            <TextField
            type='password'
            name='password'
            label='Пароль'
            error={this.state.error}
            value={this.state.password}
            margin='normal'
            variant='outlined'
            onChange={this.changeHandler}
            placeholder='Введите ваш пароль'
            />
          </FormGroup>
          <MySubmitButton />
        </form>
      </MyContainer>
    )
  }
}

const mapStateToProps = (state) => ({
  auth: state.auth.payload,
  error: state.error.payload
})

export default connect(mapStateToProps)(Login)