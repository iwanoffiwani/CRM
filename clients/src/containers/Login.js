import React, { useState } from 'react'
import { connect } from 'react-redux'
import { makeStyles } from '@material-ui/core/styles'
import { loginUser } from '../redux/actions'
import Container from '@material-ui/core/Container'
import Typography from '@material-ui/core/Typography'
import FormGroup from '@material-ui/core/FormGroup'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'

const useStyles = makeStyles({
  container: {
    display: 'flex',
    minHeight:'100vh',
    alignItems: 'center',
    marginTop: '-5%'
  },
  submit: {
    marginTop: '2em'
  }
})

export const Login = props => {

  const classes = useStyles()

  const { error, message } = props

  const initialState = {
    user: {
      data: {
        login: '',
        password: ''
      }
    }
  }

  const [ state, setState ] = useState(initialState)

  const changeLoginHandler = e => {
    return setState({
      ...state,
      user: {
        ...state.user,
        data: {
          ...state.user.data,
          login: e.target.value
        }
      }
    })
  }

  const changePasswordHandler = e => {
    return setState({
      ...state,
      user: {
        ...state.user,
        data: {
          ...state.user.data,
          password: e.target.value
        }
      }
    })
  }

  const submitHandler = e => {
    e.preventDefault()

    const { login, password } = state.user.data
    props.authorization({ login, password }, props.history)

    return setState({
      ...initialState
    })
  }

  return (
    <Container maxWidth='xs' className={classes.container}>
      <form style={{ width: '100%' }} onSubmit={submitHandler}>
        {
          error ? 
          <Typography
          color='error'
          variant='caption'
          >{message}</Typography> : false
        }
        <Typography
        color='textPrimary'
        variant='h5'
        >Вход в систему</Typography>
        <FormGroup>
          <TextField
          type='text'
          name='login'
          label='Логин'
          error={error}
          value={state.user.data.login}
          margin='normal'
          variant='outlined'
          onChange={changeLoginHandler}
          placeholder='Введите ваш логин'
          />
        </FormGroup>
        <FormGroup>
          <TextField
          type='password'
          name='password'
          label='Пароль'
          error={error}
          value={state.user.data.password}
          margin='normal'
          variant='outlined'
          onChange={changePasswordHandler}
          placeholder='Введите ваш пароль'
          />
        </FormGroup>
        <Button
        type='submit'
        size='medium'
        color='primary'
        variant='contained'
        className={classes.submit}
        >Войти</Button>
      </form>
    </Container>
  )
}

const mapStateToProps = state => {
  return {
    error: state.authorization.payload.error,
    message: state.authorization.payload.message
  }
}

const mapDispatchToProps = dispatch => {
  return {
    authorization: (user, history) => dispatch(loginUser(user, history))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Login)