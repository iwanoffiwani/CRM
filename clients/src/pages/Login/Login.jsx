import React from 'react'
import Auth from '../../components/Auth/Auth'

const PageAuthorization = () => (
  <Auth />
)

export default PageAuthorization