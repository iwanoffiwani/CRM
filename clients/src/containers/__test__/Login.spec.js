import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Login } from '../Login'

describe('Testing the Login component', () => {
  it('+++ renderer component without props', () => {
    const wrapper = shallow(<Login />)
    let tree = toJson(wrapper)

    return expect(tree).toMatchSnapshot()
  })

  it('+++ renderer component with error message', () => {
    const nextProps = {
      error: true,
      message: 'Some text for error message'
    }

    const wrapper = shallow(<Login {...nextProps} />)
    const tree = toJson(wrapper)

    expect(tree).toMatchSnapshot()
  })
})