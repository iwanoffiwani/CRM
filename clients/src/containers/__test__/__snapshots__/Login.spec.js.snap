// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Login component +++ renderer component with error message 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-container-1"
  maxWidth="xs"
>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      color="error"
      variant="caption"
    >
      Some text for error message
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textPrimary"
      variant="h5"
    >
      Вход в систему
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(FormGroup))>
      <WithStyles(ForwardRef(TextField))
        error={true}
        label="Логин"
        margin="normal"
        name="login"
        onChange={[Function]}
        placeholder="Введите ваш логин"
        type="text"
        value=""
        variant="outlined"
      />
    </WithStyles(ForwardRef(FormGroup))>
    <WithStyles(ForwardRef(FormGroup))>
      <WithStyles(ForwardRef(TextField))
        error={true}
        label="Пароль"
        margin="normal"
        name="password"
        onChange={[Function]}
        placeholder="Введите ваш пароль"
        type="password"
        value=""
        variant="outlined"
      />
    </WithStyles(ForwardRef(FormGroup))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-submit-2"
      color="primary"
      size="medium"
      type="submit"
      variant="contained"
    >
      Войти
    </WithStyles(ForwardRef(Button))>
  </form>
</WithStyles(ForwardRef(Container))>
`;

exports[`Testing the Login component +++ renderer component without props 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-container-1"
  maxWidth="xs"
>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      color="textPrimary"
      variant="h5"
    >
      Вход в систему
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(FormGroup))>
      <WithStyles(ForwardRef(TextField))
        label="Логин"
        margin="normal"
        name="login"
        onChange={[Function]}
        placeholder="Введите ваш логин"
        type="text"
        value=""
        variant="outlined"
      />
    </WithStyles(ForwardRef(FormGroup))>
    <WithStyles(ForwardRef(FormGroup))>
      <WithStyles(ForwardRef(TextField))
        label="Пароль"
        margin="normal"
        name="password"
        onChange={[Function]}
        placeholder="Введите ваш пароль"
        type="password"
        value=""
        variant="outlined"
      />
    </WithStyles(ForwardRef(FormGroup))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-submit-2"
      color="primary"
      size="medium"
      type="submit"
      variant="contained"
    >
      Войти
    </WithStyles(ForwardRef(Button))>
  </form>
</WithStyles(ForwardRef(Container))>
`;
